lasagne/layers/pool.py

```diff
@@ -217,28 +229,44 @@ def get_output_for(self, input, **kwargs):
 
 
 class FeaturePoolLayer(Layer):
-
     """
-    This layer performs pooling across feature maps, which can be used
-    to implement maxout.
-    IMPORTANT: this layer requires that the number of feature maps is
-    a multiple of the pool size.
+    Feature pooling layer
+
+    This layer pools across a given axis of the input. By default this is axis
+    1, which corresponds to the feature axis for :class:`DenseLayer`,
+    :class:`Conv1DLayer` and :class:`Conv2DLayer`. The layer can be used to
+    implement maxout.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or tuple
+        The layer feeding into this layer, or the expected input shape.
+
+    pool_size : integer
+        the size of the pooling windows, i.e. the number of features / feature
```

Comment:
Did I hear "name discussion"? :)

"groups" are used in cuda-convnet for grouped convolution, where you learn kernels for subsets of the input channels. In this case, you don't specify the size of the pooling region/window/thing, but the number of groups, and if you use the same number for each convolutional layer, you get a multi-column architecture.
So this notion of "groups" is closely related to what we're doing here if axis=1, but as it doesn't have to be axis=1 and if we want to have a common term, I think "region" is a good choice. "window" is something I would associate with the last axis/axes in bc01* notation.